File: Terminplaner-Java/src/Teilnahme.java
public class Teilnahme implements Comparable {

	private final String _teilnehmer;
	private TeilnahmeStatus _status = TeilnahmeStatus.OFFEN;
	
	public Teilnahme(String teilnehmer) {
		this(TeilnahmeStatus.OFFEN, teilnehmer);
	}

	public Teilnahme(TeilnahmeStatus status, String teilnehmer) {
		_status = status;
		_teilnehmer = teilnehmer;
	}

	public String getTeilnehmer() {
		return _teilnehmer;
	}

	public int compareTo(Object o) {
		return getTeilnehmer().compareTo( ((Teilnahme)o).getTeilnehmer() );
	}

	public int hashCode() {
		return getTeilnehmer().hashCode();
	}
	
	
	public boolean equals (Object o) {
		return getTeilnehmer().equals( ((Teilnahme)o).getTeilnehmer() );
	}

	public TeilnahmeStatus getStatus() {
		return _status;
	}

	public void bestaetige() {
		_status = TeilnahmeStatus.BESTAETIGT;
	}

	public void lehneAb() {
		_status = TeilnahmeStatus.ABGELEHNT;
	}

	public boolean istAbgelehnt() {
		return _status.equals(TeilnahmeStatus.ABGELEHNT);
	}

}

File: Terminplaner-Ruby/src/Teilnahme.rb
class Teilnahme 

	include Comparable

  	@@OFFEN = 1
	@@BESTAETIGT = 2
	@@ABGELEHNT = 3	

	attr_reader :teilnehmer, :status

	def Teilnahme::new_default(teilnehmer)
		Teilnahme.new(@@OFFEN, teilnehmer)
	end

	def initialize(status, teilnehmer)
		@status = status
		@teilnehmer = teilnehmer
	end

	def <=> (obj)
		@teilnehmer <=>  obj.teilnehmer
	end

	def hash 
		@teilnehmer.hash
	end
	
	
	def == (obj) 
		@teilnehmer == obj.teilnehmer
	end

	def bestaetige
		@status = @@BESTAETIGT
	end

	def lehne_ab
		@status = @@ABGELEHNT
	end

	def ist_abgelehnt
		@status == @@ABGELEHNT
	end

	def Teilnahme::BESTAETIGT
		@@BESTAETIGT
	end


	def Teilnahme::OFFEN
		@@OFFEN
	end

	def Teilnahme::ABGELEHNT
		@@ABGELEHNT
	end

end

